### PR TITLE
Учебный проект: разделяй и властвуй (Часть 2)

### PR DESCRIPTION
**Pull request by katasonova**
---
:mortar_board: [Учебный проект: разделяй и властвуй (Часть 2)](https://up.htmlacademy.ru/ecmascript/10/user/579401/tasks/14)

:boom: https://htmlacademy-ecmascript.github.io/579401-taskmanager-10/7/

### CODE REVIEW

File: src/controllers/board.js [hunk 1]
```diff
@@ -80,23 +81,18 @@ export default class BoardController {
           sortedTasks = tasks;
           break;
       }
+      renderTasks(boardListElement, sortedTasks);
+      render(container, this._loadMoreButton.getElement());
+      this._loadMoreButton.setLoadMoreButtonClickHandler(() => loadMoreButtonClickHandler(sortedTasks));
```
Comment (katasonova) [inline src/controllers/board.js]: sorting is fixed, but the button handler behaves weirdly,
it works fine just first time, after sorting it renders everything and the order is incorrect

Comment (fyvfyv) [reply src/controllers/board.js]: You can check how many handlers you add to this button

Comment (katasonova) [reply src/controllers/board.js]: I see, so basically, every time i click sort a handler will be added to the button.
How to fix? Do i need for button class setDeleteButtonHandler()? or is there a different approach?

Comment (fyvfyv) [reply src/controllers/board.js]: You don't need to add it every time.

Comment (katasonova) [reply src/controllers/board.js]: ok, cool, it's fixed!

but i don't get one thing it's all sorted ok, but when i press the button to load more the newly loaded tasks aren't sorted :(

Comment (fyvfyv) [reply src/controllers/board.js]: exhausted you sort separately every page

Comment (katasonova) [reply src/controllers/board.js]: yeah, but i can't change that, can i?
it sorted separately because of the different sorting options

Comment (fyvfyv) [reply src/controllers/board.js]: https://github.com/htmlacademy-ecmascript/taskmanager-10/commit/31c55566f8d030a622b5d51dbc3e6f44f547181c#r36479904

Comment (katasonova) [reply src/controllers/board.js]: ok, but how to do that? - sort everything and show it per page, i can't think of any implementation

Comment (fyvfyv) [reply src/controllers/board.js]: You take whole array, sort it, and «show more» button only show next part of this whole array

**Pull request merged**